addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -0,0 +1,764 @@
+
+    def _post(self, soft=True):
+        # EXTEND 'account'
+        posted = super()._post(soft=soft)
+
+        # Ensure to tag the move as 'Ready to send' upon posting if it makes sense.
+        posted.filtered(
+            lambda invoice:
+                invoice.country_code == 'VN'
+                and invoice.is_sale_document()
+                and not invoice._l10n_vn_edi_is_sent()
+        ).l10n_vn_edi_invoice_state = 'ready_to_send'
+
+        return posted
+
+    # -------------------------------------------------------------------------
+    # API METHODS
+    # -------------------------------------------------------------------------
+
+    def _l10n_vn_edi_check_invoice_configuration(self):
+        """ Some checks that are used to avoid common errors before sending the invoice. """
+        self.ensure_one()
+        company = self.company_id
+        commercial_partner = self.commercial_partner_id
+        errors = []
+        if not company.l10n_vn_edi_username or not company.l10n_vn_edi_password:
+            errors.append(_('Sinvoice credentials are missing on company %s.', company.display_name))
+        if not company.vat:
+            errors.append(_('VAT number is missing on company %s.', company.display_name))
+        company_phone = company.phone and self._l10n_vn_edi_format_phone_number(company.phone)
+        if company_phone and not company_phone.isdecimal():
+            errors.append(_('Phone number for company %s must only contain digits or +.', company.display_name))
+        commercial_partner_phone = commercial_partner.phone and self._l10n_vn_edi_format_phone_number(commercial_partner.phone)
+        if commercial_partner_phone and not commercial_partner_phone.isdecimal():
+            errors.append(_('Phone number for partner %s must only contain digits or +.', commercial_partner.display_name))
+        if not self.l10n_vn_edi_invoice_symbol:
+            errors.append(_('The invoice symbol must be provided.'))
+        if self.l10n_vn_edi_invoice_symbol and not self.l10n_vn_edi_invoice_symbol.invoice_template_id:
+            errors.append(_("The invoice symbol's template must be provided."))
+        if self.move_type == 'out_refund' and (not self.reversed_entry_id or not self.reversed_entry_id._l10n_vn_edi_is_sent()):
+            errors.append(_('You can only send a credit note linked to a previously sent invoice.'))
+        if not self.partner_id.street or not self.partner_id.city or not self.partner_id.state_id or not self.partner_id.country_id:
+            errors.append(_('The street, city, state and country of partner %s must be provided.', self.partner_id.display_name))
+        if not company.street or not company.state_id or not company.country_id:
+            errors.append(_('The street, state and country of company %s must be provided.', company.display_name))
+        vnd = self.env.ref('base.VND')
+        if self.company_currency_id.id != vnd.id:
```

Comment:
Hm sure, but the next line I'm using the currency record to check the rate.
I guess it's not the most important check and could be removed, although it could avoid some mishaps down the line.